hudi-client/hudi-spark-client/src/main/scala/org/apache/spark/sql/hudi/execution/RangeSample.scala

```diff
@@ -316,6 +316,8 @@ object RangeSampleSort {
         HoodieClusteringConfig.LAYOUT_OPTIMIZE_BUILD_CURVE_SAMPLE_SIZE.defaultValue.toString).toInt
       val sample = new RangeSample(zOrderBounds, sampleRdd)
       val rangeBounds = sample.getRangeBounds()
+      if (rangeBounds.size <= 1)
```

Comment:
Can also fix `sortDataFrameBySampleSupportAllTypes`, it should have the same issue

Reply:
likely this one was not failing, still added a short path to avoid sorting

Reply:
if sort columns contain complex type, `sortDataFrameBySampleSupportAllTypes` will be used

Reply:
yes, the test has `height` column which is complex (array). But it did't trigger an error, a simple columns did.